mrmustard/lab/representations/data/symplectic_data.py

```diff
@@ -0,0 +1,64 @@
+# Copyright 2023 Xanadu Quantum Technologies Inc.
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from __future__ import annotations
+
+import numpy as np
+
+from mrmustard.lab.representations.data.matvec_data import MatVecData
+from mrmustard.math import Math
+from mrmustard.typing import RealMatrix, Scalar, RealVector
+from thewalrus.quantum.gaussian_checks import is_symplectic
+
+math = Math()
+
+
+class SymplecticData(MatVecData):
+    """Symplectic matrix-like data for certain Representation objects.
+
+    Here the displacement vector is defined as:
+    :math:`\bm{d} = \sqrt{2\hbar}[\Re(\alpha), \Im(\alpha)]`
+
+    Args:
+        symplectic:     symplectic matrix with qqpp-ordering
+        displacement:   the real displacement vector
+        coeffs:         default to be 1.
+    """
+
+    def __init__(
+        self, symplectic: RealMatrix, displacement: RealVector, coeffs: Scalar = 1.0
+    ) -> None:
+        # Check if it is a symplectic matrix
+        if is_symplectic(math.asnumpy(symplectic)):
+            super().__init__(mat=symplectic, vec=displacement, coeffs=coeffs)
+        else:
+            raise ValueError("The matrix given is not symplectic.")
+
+    @property
+    def symplectic(self) -> np.array:
+        return self.mat
+
+    @property
+    def displacement(self) -> np.array:
+        return self.vec
+
+    def __mul__(self, other: Scalar) -> SymplecticData:
+        if isinstance(other, SymplecticData):
+            raise TypeError("Symplectic can only be multiplied by a scalar")
```

Comment:
"We do not support multiplication of two symplectic data now."

Reply:
For symplectic data, no

Reply:
Ok. So I'm not sure why the current error message is wrong. ^^'

Reply:
Ah it is not wrong. Just provide another way to say the sentence. That's why I use "" to quote.